server/src/main/java/org/elasticsearch/action/NoShardAvailableActionException.java

```diff
@@ -37,5 +55,31 @@ public RestStatus status() {
 
     public NoShardAvailableActionException(StreamInput in) throws IOException {
         super(in);
+        if (in.getVersion().onOrAfter(Version.V_8_7_0)) {
+            this.onShardFailureWrapper = in.readBoolean();
+        } else {
+            this.onShardFailureWrapper = false;
```

Comment:
I was thinking that the flag does not need to be serialized (we could mark it transient to signal that for instance). I may be missing something, but ideally we set the flag when we throw the exception, to make sure that on that node the stacktrace will be seen as empty, which ensures that an empty array gets sent over the wire. On the other side, the stacktrace array is received empty, and there is no need for the flag to be set, false is fine too in that case because there is no stacktrace to show. Does this make sense though?

Reply:
You are correct! We don't need to serialize field. I will remove that.

We only need to worry about the stack trace being empty on serialization, then the node reading in will simply read an empty array, and this works just fine.